### PR TITLE
PolyMC: Update Java runtime dependencies

### LINKED ISSUE #38645: PolyMC Depends On Old Version of Java
### Is this a new report?

Yes

### System Info

Void 5.18.17_1 x86_64 GenuineIntel uptodate FFFFF

### Package(s) Affected

PolyMC-1.4.1_1

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

_No response_

### Expected behaviour

The package recommends a version of Java compatible with the latest version of Minecraft on installation. After this, when launching a profile with the latest version of Minecraft, there is no error or need to manually install a later version of Java via `xbps`

### Actual behaviour

The package recommends Java version 11, far from the necessary versions 17 or 18 for running a modern version of Minecraft. I'll see about creating a PR to change this.

### Steps to reproduce

1. Install the package
2. Start the application
3. Set a Java runtime (default being 11)
4. Set up a Minecraft profile and launch
5. Note the launch fails requesting Java version 17 or 18

### PR DESCRIPTION
**Pull request by TrueTechie**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64_glibc

Fixes #38645

### CODE REVIEW

File: srcpkgs/PolyMC/template [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+revision=2
 build_style=cmake
 configure_args="-DLauncher_BUILD_PLATFORM=Void
  -DLauncher_VERSION_BUILD=${revision}"
-hostmakedepends="extra-cmake-modules openjdk8 pkg-config qt5-host-tools
+hostmakedepends="extra-cmake-modules openjdk17 pkg-config qt5-host-tools
  qt5-qmake scdoc"
 makedepends="qt5-devel"
-depends="virtual?java-runtime qt5-svg qt5-imageformats xrandr"
+depends="openjdk17-jre qt5-svg qt5-imageformats xrandr"
```
Comment (JamiKettunen) [inline srcpkgs/PolyMC/template:11]: Why? `virtual?java-runtime` will accept any package that provides `java-runtime` virtual dependency, this includes both `openjdk17-jre` and the [current default of `openjdk8-jre`](https://github.com/void-linux/void-packages/blob/4475c34/etc/defaults.virtual#L25).

If you install `openjdk17-jre` you should be able to remove `openjdk8-jre`, or why not keep both. You could also install `openjdk17-jre` before `PolyMC` and then `openjdk8-jre` won't be pulled in.

Comment (JamiKettunen) [reply srcpkgs/PolyMC/template:11]: Ah nvm attempting to install `PolyMC` pulls in `openjdk11-jre` by default for me on `x86_64-musl`, wonder what exactly makes that the case... Either way this is a bit of a nuanced issue, perhaps @classabbyamp can comment better.

I previously suggested [mentioning the existance of PolyMC's README.voidlinux](https://github.com/void-linux/void-packages/pull/34979#issuecomment-1079054056) but it seems that was completely lost in the comments, but the file should already exist as [`/usr/share/doc/steam/README.voidlinux`](https://github.com/void-linux/void-packages/blob/master/srcpkgs/PolyMC/files/README.voidlinux) on your system.

**Comment by paper42** [comment]
Different versions of minecraft require different versions of java. We don't want to require all of them, so I think we should keep using the virtual dependency that can be satisfied by jre 8, 11 or 17.

File: srcpkgs/PolyMC/template (hunk 1 above)
Comment (TrueTechie) [reply srcpkgs/PolyMC/template:11]: > Ah nvm attempting to install `PolyMC` pulls in `openjdk11-jre` by default for me on `x86_64-musl`, wonder what exactly makes that the case... Either way this is a bit of a nuanced issue, perhaps @classabbyamp can comment better.
> 
> I previously suggested [mentioning the existance of PolyMC's README.voidlinux](https://github.com/void-linux/void-packages/pull/34979#issuecomment-1079054056) but it seems that was completely lost in the comments, but the file should already exist as [`/usr/share/doc/PolyMC/README.voidlinux`](https://github.com/void-linux/void-packages/blob/master/srcpkgs/PolyMC/files/README.voidlinux) on your system.

Fair, but README or not, this still means that the package wouldn't have accurate dependencies which puts into question the point of deps in the first place. Not to mention the extra work for new users who'll have to switch one version out for another.

**Comment by TrueTechie** [comment]
> Different versions of minecraft require different versions of java. We don't want to require all of them, so I think we should keep using the virtual dependency that can be satisfied by jre 8, 11 or 17.

Fair, but having specifically the up-to-date version of the game that most people want to play be the one that doesn't work. If it's a later version of Java by default, less players have to go out of their way to change it manually.

**Comment by classabbyamp** [comment]
Void policy is to only include the *absolutely required* software in dependencies, not optional software. PolyMC can work with any of the three openjdk runtimes we currently package, so that requirement is put in the depends.

PolyMC provides a nice, clear error message if it doesn't have the correct jdk *and* README.voidlinux (which is [the standard way](https://docs.voidlinux.org/config/package-documentation/index.html) to let users know about void-specific documentation with a package) says the same thing.

Forcing the dependency to `openjdk17-jre` would cause errors if a user only wanted to have other versions of openjdk on their system:
```
xbps-install PolyMC
...
xbps-remove openjdk17-jre
openjdk17-jre-17.0.3+7_1 in transaction breaks installed pkg `PolyMC-1.4.1_2'
```
This is not good.

`virtual?openjdk-runtime` is the correct dependency for PolyMC.

**Pull request closed without merging**